plugins/maciVoting/contexts/types.ts

```diff
@@ -1,11 +1,33 @@
 import { type Keypair } from "@maci-protocol/domainobjs";
+import { type z } from "zod";
 import { type VoteOption } from "../utils/types";
+import { type GenerateResponseSchema, type SubmitResponseSchema } from "./schemas";
 
 export interface IVoteArgs {
   voteOptionIndex: bigint;
   newVoteWeight: bigint;
 }
 
+export type GenerateResponse = z.infer<typeof GenerateResponseSchema>;
+export type SubmitResponse = z.infer<typeof SubmitResponseSchema>;
+export type CoordinatorServiceResult<T, E = Error> = { success: true; data: T } | { success: false; error: E };
+
+export interface CoordinatorContextType {
+  merge: (pollId: number) => Promise<CoordinatorServiceResult<boolean>>;
+  generateProofs: ({
+    pollId,
+    encryptedCoordinatorPrivateKey,
+    startBlock,
+    endBlock,
+  }: {
+    pollId: number;
+    encryptedCoordinatorPrivateKey: string;
+    startBlock: number;
+    endBlock: number;
+  }) => Promise<CoordinatorServiceResult<GenerateResponse>>;
```

Comment:
Let's move it to the interface and reuse

Reply:
Done